blackbird_python/blackbird/auxiliary.py

```diff
@@ -208,7 +208,7 @@ def _expression(expr):
         return _VAR[expr.NAME().getText()].flatten()[inner_expr]
 
     if isinstance(expr, blackbirdParser.ParameterLabelContext):
-        p = Symbol(expr.NAME().getText())
+        p = Symbol(expr.parameter().NAME().getText())
```

Comment:
How come this had to be changed?

Reply:
It's because the grammar was changed to allow for using parameters in arrays. I created a new `parameter` rule that needs to be called as well. You can see this in the `blackbird.g4` file. :slightly_smiling_face:

Reply:
Right! One thing missing for me is that this would be modifying an existing rule and was not quite sure why now calling the `parameter` method was necessary :thinking:

Reply:
It's just the way ANTLR works. Before a `ParameterLabelContext` was defined as part of an `expression` directly:
```
LBRACE NAME RBRACE;   #ParameterLabel
```
but now it's as a `parameter`:
```
parameter             #ParameterLabel
```
where `parameter` is:
```
parameter : LBRACE NAME RBRACE;
```
because I wanted to use `parameter` in the `array` rules as well.

Reply:
Ohh, got ya! Was missing that the `parameter` method was freshly generated here. Thanks for the explanation!